### PR TITLE
[NETBEANS-1278]:Schema2Bean doesn't support remote schemas and DTDs

### PR DESCRIPTION
**Pull request by ehsavoie**
* Adding support for remote schemas and dtds
 * fixing return value if the type is primitive (it was returning null)
 * Adding support for remote xsd import

Jira: https://issues.apache.org/jira/browse/NETBEANS-1278

### CODE REVIEW

**Comment by geertjanw** [comment]
+1 Cool.

**Comment by matthiasblaesing** [comment]
You claim, that we can't integrate the schema files for JEE - which ones can't we include? The standard ones are CDDL and so can be integrated (and in fact are already at least partly). From my POV: If we can, we should ship the Schema files, as network roundtrips are a problem performance and security wise. I agree, that Wildfly/JBoss support might be a problem, but that raises the question: When code is generated from an LGPL/GPL XSD, what is the license of the generated code?

I would be good, if you could separate the commit with regard to the issue to solve. In your description you mention three distinct changes, seeing the change separately might make it easier to spot the change.

**Comment by ehsavoie** [comment]
@matthiasblaesing I'll create a JIRA for the legal part but according to https://www.apache.org/legal/resolved.html we can't distribute the LGPL schema and I'm not sure we could move them to ASL 2.0 easily and quickly enough.
I can split the commits if you want but you can't have the 1st item without the 3nd (there are bound) and the 2nd is just a quick fix (https://github.com/apache/incubator-netbeans/pull/895/commits/774361396d8e7d6f6243d61f514cbf118f106335#diff-74c86da3c52cf7750ce9c7e78e513660R862)

**Comment by ehsavoie** [comment]
Splitted the initial commit in 3 parts

File: ide/schema2beans/src/org/netbeans/modules/schema2beansdev/BeanClass.java [hunk 1]
```diff
@@ -859,7 +859,11 @@ void genAccessors(int out) throws IOException {
                 AttrProp attrProp = a.getAttrProp();
                 jw.comment("If our element does not exist, then the attribute does not exist.");
                 jw.beginIf("size("+attributeOwner.constName+") == 0");
-                jw.writeEol("return null");
+                if("boolean".equals(attrProp.getJavaType()) || "int".equals(attrProp.getJavaType()) || "double".equals(attrProp.getJavaType())) {
+                    jw.writeEol("return " + attrProp.getDefaultValue());
+                } else {
+                    jw.writeEol("return null");
+                }
```
[the remark this answers is not in the document]
Comment (ehsavoie) [reply ide/schema2beans/src/org/netbeans/modules/schema2beansdev/BeanClass.java]: Question1 : I wish I knew
Question2 : I asked myself the same question but it wasn't breaking my code while the boolean was so I kept the changes to minimal but I agree we should cover them

**Comment by matthiasblaesing** [comment]
@ehsavoie two things:
1. we established, that generating code from remote schemas won't help us for our license problem. I would strongly suggest not implement remote schema loading, as this opens a whole new can of worms.
2. If 1. is accepted, then the change in attribute handling remains. Could you give me an example for an schema (a minimal sample would be great), that fails to parse? I was not able to create the broken code...

**Comment by matthiasblaesing** [comment]
Closing this as it got stale.

**Pull request closed without merging**